Description: [PATCH] x11: Protect XChangeProperty call with error traps
 When closing maximized window of any GTK app gnome-shell and
 budgie desktop crashes and restarts. This issue specifically affects
 users of the gnome-shell xorg session together with all budgie desktop
 users.
Author: polygamma <jonny.westphalen@googlemail.com>
Origin: upstream commit f9c625924e5358afd3e47733e68d166770edd496
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=788666
Bug-Ubuntu: https://launchpad.net/bugs/1722510
Last-Update: 2017-10-09

---
 src/x11/window-x11.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/x11/window-x11.c b/src/x11/window-x11.c
index 36a5e70a3..4885f5fea 100644
--- a/src/x11/window-x11.c
+++ b/src/x11/window-x11.c
@@ -920,11 +920,13 @@ update_gtk_edge_constraints (MetaWindow *window)
 
   meta_verbose ("Setting _GTK_EDGE_CONSTRAINTS to %lu\n", data[0]);
 
+  meta_error_trap_push (window->display);
   XChangeProperty (window->display->xdisplay,
                    window->xwindow,
                    window->display->atom__GTK_EDGE_CONSTRAINTS,
                    XA_CARDINAL, 32, PropModeReplace,
                    (guchar*) data, 1);
+  meta_error_trap_pop (window->display);
 }
 
 static gboolean
